src/components/ReportViewer/multi-select-list.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import React, { useMemo, useState } from 'react';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import { Box, Checkbox, FormControlLabel, IconButton } from '@mui/material';
+import MenuIcon from '@mui/icons-material/Menu';
+
+const styles = {
+    notificationDot: {
+        height: '6px',
+        width: '6px',
+        backgroundColor: '#cc70a0',
+        borderRadius: '50%',
+        position: 'absolute',
+        top: '5px',
+        left: '23px',
+    },
+    container: {
+        position: 'relative',
+    },
+    icon: {
+        width: '0.7em',
+        height: '0.7em',
+    },
+    label: {
+        width: '100%',
+        margin: 0,
+    },
+    menuItem: {
+        padding: '0 10px 0 0',
+    },
+};
+
+/**
+ * MultiSelectList allows to manipulate an object where each keys are associated to a boolean in order to determine which are the ones the user wants to select
+ * A visual indication is displayed when the user alters the default state of the list
+ *
+ * @param {Object}   selectedItems    - It serves as the model and data of the Component each entry must be formatted as a pair of string and boolean. Each key will be displayed and the corresponding boolean is updated in function of its checkbox status
+ * @param {Function} setSelectedItems - Setter needed to update the list underlying data
+ */
+
+export const MultiSelectList = ({ selectedItems, setSelectedItems }) => {
```

Comment:
it could be nice to dissociate the MultiSelectList, which is a component use to select multiple elements, and the filterElement, which is a clickable button, which displays the MultiSelectList when clicked

if we need another type of filter, it will be really easier to develop
and it will make both components easier to maintain / understand

Reply:
Agreed, I've relocated the button part in a new component